podpac/datalib/smap.py

```diff
@@ -190,29 +196,39 @@ def _infer_SMAP_product_version(product, base_url, auth_session):
 
 # Discover SMAP OpenDAP url from podpac s3 server
 SMAP_BASE_URL_FILE = os.path.join(os.path.dirname(__file__), 'nsidc_smap_opendap_url.txt')
-SMAP_BASE_URL = 'https://n5eil01u.ecs.nsidc.org/opendap/SMAP'
-try:
-    with open(SMAP_BASE_URL_FILE, 'r') as fid:
-        rf = fid.read()
-    if 'https://' in rf and 'nsidc.org' in rf:
-        SMAP_BASE_URL = rf
-except Exception as e:
-    warnings.warn("Could not retrieve SMAP url from %s: " % (SMAP_BASE_URL_FILE) + str(e))
-try:
-    r = requests.get('https://s3.amazonaws.com/podpac-s3/settings/nsidc_smap_opendap_url.txt').text
-    if 'https://' in r and 'nsidc.org' in r:
-        if rf != r:
-            warnings.warn("Updating SMAP url from PODPAC S3 Server.")
-            SMAP_BASE_URL = r
-            try:
-                with open(SMAP_BASE_URL_FILE, 'w') as fid:
-                    fid.write(r)
-            except Exception as e:
-                warnings.warn("Could not overwrite SMAP url update on disk:" + str(e))
-except Exception as e:
-    warnings.warn("Could not retrieve SMAP url from PODPAC S3 Server. Using default." + str(e))
-
-SMAP_BASE_URL_REGEX = re.compile(re.sub(r'\d', r'\\d', SMAP_BASE_URL.split('/')[2]))
+_SMAP_BASE_URL = None
+def SMAP_BASE_URL():
```

Comment:
I would have loved to get rid of the `()`, but I don't know the syntax for module-level properties. Anyway, lazy-loading makes importing podpac faster.